docs/running-on-kubernetes.md

```diff
@@ -111,70 +162,96 @@ If our local proxy were listening on port 8001, we would have our submission loo
-namespace (set by `spark.kubernetes.namespace`). The process should determine a URI that routes to this service
-(potentially configuring infrastructure to handle the URI behind the scenes), and patch the service to include an
-annotation `spark-job.alpha.apache.org/resolvedExternalUri`, which has its value as the external URI that your process
-has provided (e.g. `https://example.com:8080/my-job`).
-
-Note that the URI provided in the annotation needs to route traffic to the appropriate destination on the pod, which has
-a empty path portion of the URI. This means the external URI provider will likely need to rewrite the path from the
-external URI to the destination on the pod, e.g. https://example.com:8080/spark-app-1/submit will need to route traffic
-to https://<pod_ip>:<service_port>/. Note that the paths of these two URLs are different.
-
-If the above is confusing, keep in mind that this functionality is only necessary if the submitter cannot reach any of
-the nodes at the driver's node port. It is recommended to use the default configuration with the node port service
-whenever possible.
+
+### Securing the Resource Staging Server with TLS
+
+The default configuration of the resource staging server is not secured with TLS. It is highly recommended to configure
+this.
+
+The YAML file in `conf/kubernetes-resource-staging-server.yaml` includes a ConfigMap resource that holds the resource
+staging server's configuration. The properties can be adjusted here to make the resource staging server listen over TLS.
+Refer to the [security](security.html) page for the available settings related to TLS. The namespace for the
+resource staging server is `kubernetes.resourceStagingServer`, so for example the path to the server's keyStore would
+be set by `spark.ssl.kubernetes.resourceStagingServer.keyStore`.
+
+In addition to the settings specified by the previously linked security page, the resource staging server supports the
+following additional configurations:
+
+- `spark.ssl.kubernetes.resourceStagingServer.keyPem` and `spark.ssl.kubernetes.resourceStagingServer.serverCertPem`, to
+use PEM-encoded keys and certificates instead of Java keyStore files. Note that both of these must be specified
+simultaneously. If these are provided, then `spark.ssl.kubernetes.resourceStagingServer.keyStore` must not be specified.
+- `spark.ssl.kubernetes.resourceStagingServer.keyStorePasswordFile`, to provide the keyStore's password using a file
+instead of a static value. This is useful if the keyStore's password is to be mounted into the container with a secret.
+- `spark.ssl.kubernetes.resourceStagingServer.keyPasswordFile`, to provide the keyStore's key password using a file
+instead of a static value. This is useful if the key password is to be mounted into the container with a secret.
+
+Note that while the properties can be set in the ConfigMap, you will still need to consider the means of mounting the
```

Comment:
@foxish I wonder if we can do better here. Perhaps it would be worthwhile to have a script that allows users to provide their TLS secrets from their local disk and the script then converts them to a Kubernetes secret that's mounted into the pod. Or if that's too heavyweight and we don't want to maintain that, perhaps we can have an alternate YML template that provides an outline for a Secret object and how it would be mounted into the resource staging service container.